crates/sui-adapter/src/temporary_store.rs

```diff
@@ -349,12 +366,56 @@ impl<S> Storage for TemporaryStore<S> {
 
         // For object deletion, we increment their version so that they will
         // eventually show up in the parent_sync table with an updated version.
-        self.deleted.insert(*id, (version.increment(), kind));
+        self.deleted
+            .insert(*id, (version.increment(), child_count, kind));
     }
 
     fn log_event(&mut self, event: Event) {
         self.events.push(event)
     }
+
+    fn change_child_counts(&mut self, deltas: BTreeMap<ObjectID, i64>) -> Result<(), ObjectID> {
+        let deltas = deltas
+            .into_iter()
+            .filter(|(_, delta)| *delta != 0)
+            .collect::<BTreeMap<_, _>>();
+        for id in deltas.keys() {
+            if !(self.written.contains_key(id) || self.deleted.contains_key(id)) {
+                let mut object = self.objects[id].clone();
+                // Active input object must be Move object.
+                object.data.try_as_move_mut().unwrap().increment_version();
+                self.written.insert(*id, object);
```

Comment:
I don't quite understand this.
btw any object update now must be done through self.write_object() to make sure previous transaction is updated.

Reply:
I can try to do that. But it is just trying to mark the object as written if the child_count has changed

Reply:
Changed this a bit, but I'm still manually inspecting the _written map

Reply:
hmm actually, this may be an issue. So some objects may get to be written twiece

Reply:
I'm not sure I understand the concern